lib/garage/tracer.rb

```diff
@@ -0,0 +1,90 @@
+module Garage
+  module Tracer
+    extend self
+    extend Forwardable
+
+    def_delegators :tracer, :start, :inject_trace_contect, :record_http_request, :record_http_response
+
+    private
+
+    def tracer
+      case Garage.configuration.tracing[:tracer]
```

Comment:
I think users cannot use custom tracer with this interface. Users can only use aws-xray or null.
How about setting tracer object directly like below?

```ruby
Garage.configuration.tracer = Garage::Tracer::AwsXrayTracer.new
```